portfolio/src/main/webapp/script.js

```diff
@@ -28,10 +28,33 @@ function addRandomGreeting() {
 }
 
 function addRandomNameGreeting(){
-    console.log("Fetching a randome greeting with a name   ");
+    console.log("Fetching a random greeting with a name   ");
     fetch("/data").then(response => response.text()).then((name) => {
         document.getElementById('name-container').innerText = name;
     });
 
 }
 
+function postComments() {
+    fetch('/data').then(response => response.json()).then((comments) => {
+      const commentsListElement = document.getElementById('comments-container');
+        console.log(comments);
+        commentsListElement.innerHTML = '';
+        commentsListElement.appendChild(
+            createListElement('Comment ' + comments));
```

Comment:
isn't comments the whole list?  I expect you don't want to add it 3 times.  Rather you want a forEach or a loop of some kind to add each comment just once?